src/urllib3/util/url.py

```diff
@@ -61,7 +61,7 @@
 _BRACELESS_IPV6_ADDRZ_RE = re.compile("^" + _IPV6_ADDRZ_PAT[2:-2] + "$")
 _ZONE_ID_RE = re.compile("(" + _ZONE_ID_PAT + r")\]$")
 
-_HOST_PORT_PAT = ("^(%s|%s|%s)(?::([0-9]{0,5}))?$") % (
+_HOST_PORT_PAT = ("^(%s|%s|%s)(?::([0-9]*))?$") % (
```

Comment:
Bit of a warning about this. We have to restrict the size.

```
int("0"*10000 + "1")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Exceeds the limit (4300) for integer string conversion: value has 10001 digits
```

Reply:
Maybe simply removing leading zeros would do it. Nonetheless, I am not sure about what is reasonable, len-wise. <4300 for sure.

Reply:
I see, I think we can fix the regex to catch any leading zeros in the port in a non-capturing group, discarding them.